### PR TITLE
[TreeView] Replace `<TreeItem />` with `<TreeItem2 />` and migrate all the components and utils

### LINKED ISSUE #14767: [TreeView] Drop `TreeItem` and rename `TreeItem2` into `TreeItem` (same for `useTreeItem2` and all related components)
Part of #13188

The `TreeItem2` component is becoming the default and only DX to customize the Tree Item, it should not longer have a `2` suffix.

- `TreeItem2` => `TreeItem`
- `useTreeItem2` (and `unstable_useTreeItem2`) => `useTreeItem`
- `TreeItem2Icon` => `TreeItemIcon`
- `TreeItem2Provider` => `TreeItemProvider`
- `useTreeItem2Utils` => `useTreeItemUtils`
- `TreeItem2Root` and other slots => `TreeItemRoot` etc...

Check the doc to remove any mention of `TreeItem2` and any mention of the `TreeItem` API.

**Search keywords**:

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #14767

- [x] Remove `TreeItem`
- [x] Rename `TreeItem2` => `TreeItem` (same for all other components and utils  with `TreeItem2` in their name)
- [x] Update doc
- [x] Simplify the test setup
- [x] Write codemod
- [x] Write migration guide
- [ ] Write changelog highlight (waiting for the migration guide to be review)

### CODE REVIEW

File: docs/data/tree-view/tree-item-customization/tree-item-customization.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Hover over them in the demo below to see each slot.
 <!-- TBD which option is the best: interactive or image -->
 <!-- {{"demo": "CustomTreeItemDemo.js", "hideToolbar": true}} -->
 
-{{"component": "modules/components/TreeItem2Anatomy.js"}}
+{{"component": "modules/components/TreeItemAnatomy.js"}}
```
Comment (noraleonte) [inline docs/data/tree-view/tree-item-customization/tree-item-customization.md:22]: The labels on the image still have the old naming. I'm creating a follow-up to replace them 😄

**Review by noraleonte — APPROVED** [review]
LGTM! Thanks for the effort 💙 🚀

File: docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+This is a reference guide for upgrading `@mui/x-tree-view` from v7 to v8.
+
+## Start using the new release
+
+In `package.json`, change the version of the Tree View package to `^8.0.0`.
+
+```diff
+-"@mui/x-tree-view": "7.x.x",
++"@mui/x-tree-view": "^8.0.0",
```
Comment (MBilalShafi) [inline docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md:19]: Shouldn't this next in the pre-release phase like we did in earlier releases?
```tsx
"@mui/x-tree-view": "next",
```
CC @flaviendelangle

Comment (flaviendelangle) [reply docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md:19]: In the migration guide we used the exact version, not `next`: https://mui.com/x/migration/migration-data-grid-v6/#start-using-the-new-release

No strong opinion here though

Comment (MBilalShafi) [reply docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md:19]: IIRC we used to update to the exact version on stable release.
A reference original commit on alpha for the Data Grid migration guide v6->v7: https://github.com/mui/mui-x/pull/10846/files#diff-32b8ccbf3738b9aa1a6038454e185a6aea9b2c175ab3ab7f1eb2302d8b625ff2

Comment (flaviendelangle) [reply docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md:19]: OK, that explains a lot :laughing: 
I'll update the tree view and pickers migration guides then :+1:

Comment (flaviendelangle) [reply docs/data/migration/migration-tree-view-v7/migration-tree-view-v7.md:19]: #15091

**Pull request merged**